src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -69,15 +72,21 @@ private async Task<string> GetRegionAsync(ICoreLogger logger)
 
             try
             {
-                HttpResponse response = await _httpManager.SendGetAsync(_ImdsUri, Headers, logger).ConfigureAwait(false);
+                HttpResponse response = await _httpManager.SendGetAsync(ImdsUri, Headers, logger).ConfigureAwait(false);
+
+                if (response.StatusCode != HttpStatusCode.BadRequest)
+                {
+                    UpdateImdsUriApiVersionAsync(logger);
+                    response = await _httpManager.SendGetAsync(ImdsUri, Headers, logger).ConfigureAwait(false);
+                }
 
                 if (response.StatusCode != HttpStatusCode.OK)
                 {
                     throw new MsalClientException(
```

Comment:
This is not a client exception, it's a service exception. And we should include all the HTTP error details. You can probably use https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/5000f4a5624c1acd4cf1c07b25c6abb5eedf746c/src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs#L18